### PR TITLE
fix: SAML SSO - Stop relying on `invitedTo` column and use `membership` table instead

### PR DESCRIPTION
**Pull request by devkiran**
## What does this PR do?

Previously, we used `invitedTo` to determine the team user belongs to and used that team to identify the SSO (SAML and OIDC) connections. 

More context: See message on Discord https://discord.com/channels/1125390093386010654/1146753961068597308/1150888410995752960

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- Add SSO connection to an existing team
- Login via SSO

## Mandatory Tasks

- [x] Make sure you have self-reviewed the code. A decent size PR without self-review might be rejected.

### CODE REVIEW

File: packages/features/ee/sso/lib/sso.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
+import type { PrismaClient } from "@calcom/prisma";
+import { TRPCError } from "@calcom/trpc/server";
+
+import jackson from "./jackson";
+import { tenantPrefix, samlProductID } from "./saml";
+
+export const ssoTenantProduct = async (prisma: PrismaClient, email: string) => {
+  const { connectionController } = await jackson();
+
+  const memberships = await prisma.membership.findMany({
```
Comment (devkiran) [inline packages/features/ee/sso/lib/sso.ts:10]: - Find the teams the user belongs to from the membership table
- Find the sso connection for the all the teams the user belongs to
- Find the first sso connection that is enabled for the user

File: packages/features/ee/sso/lib/sso.ts [hunk 2]
```diff
@@ -0,0 +1,57 @@
+import type { PrismaClient } from "@calcom/prisma";
+import { TRPCError } from "@calcom/trpc/server";
+
+import jackson from "./jackson";
+import { tenantPrefix, samlProductID } from "./saml";
+
+export const ssoTenantProduct = async (prisma: PrismaClient, email: string) => {
+  const { connectionController } = await jackson();
+
```
Comment (devkiran) [inline packages/features/ee/sso/lib/sso.ts:9]: - Find the teams the user belongs to from the membership table
- Find the sso connection for the all the teams the user belongs to
- Find the first sso connection that is enabled for the user

File: packages/features/ee/sso/lib/sso.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      teamId: true,
+    },
+    where: {
+      accepted: true,
+      user: {
+        email,
+      },
+    },
+  });
+
```
Comment (devkiran) [inline packages/features/ee/sso/lib/sso.ts:21]: SAML/OIDC connections are stored against the team ID. That means the app needs to determine the team ID the user belongs to to fetch the proper connection and redirect them to the IdP.

**Review by sean-brydon — APPROVED** [review]
Took me longer than i'd like to admit to setup SAML 

But this woks perfect - great job

**Pull request merged**